aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -0,0 +1,417 @@
+                                                        withShapeTensor : shapeTensor
+                                                                name : nil];
+        MPSGraphTensor* indexTensor = [mpsGraph multiplicationWithPrimaryTensor :  rangeTensor
+                                                            secondaryTensor : strideTensor
+                                                                name : nil];
+        indicesTensor = [mpsGraph additionWithPrimaryTensor : indexTensor
+                                          secondaryTensor : indicesTensor
+                                              name : nil];
+      }
+      indicesTensor = [mpsGraph additionWithPrimaryTensor : indicesTensor
+                                            secondaryTensor : storageOffsetTensor
+                                                  name : nil];
+      MPSGraphTensor *reshapedInputTensor = [mpsGraph reshapeTensor:inputTensor
+                                                         withShape:@[@-1]
+                                                              name:nil];
+      MPSGraphTensor *reshapedIndicesTensor = [mpsGraph reshapeTensor:indicesTensor
+                                                 withShape:@[@-1]
+                                                      name:nil];
+      // Call gather to coalesce the needed values. Result will be of same shape as flattened indices tensor
+      MPSGraphTensor *gatheredTensor = [mpsGraph gatherWithUpdatesTensor:reshapedInputTensor
+                                                        indicesTensor:reshapedIndicesTensor
+                                                                 axis:0
+                                                      batchDimensions:0
+                                                                 name:nil];
+
+      delete[] sizeArray;
+      // Reshape the data to desired size
+      outputTensor =  [mpsGraph reshapeTensor:gatheredTensor
+                               withShapeTensor:shapeTensor
+                                          name:nil];
+  }
+  return outputTensor;
+}
+
+
+// There are few cases we need to consider:
+// Here nodes are the Tensors and the edges are the operations performed on the
+// Tensor. As a result of the operation performed we can have result as View
+// Tensor (View T) or a Non view tensor (NonView T). The difference is if its
+// mapped by the same underlying storage ptr or a new MTLBuffer was allocated.
+//                T = Tensor
+//                 ----------
+//                 | Orig T |
+//                 ---------- 
+//                /     |     \
+//             View T  View T  NonView T
+//             /      /    \      |
+//            View T /      \     |
+//            |     /        \    |
+//            |    /          \   |
+//            |   /            \  |
+//            NonView T         NonView T
+//
+//
+Tensor as_strided_tensorimpl_mps(const Tensor& self, IntArrayRef size,
```

Comment:
@bdhirsh , this is the implementation of as_strided. We essentially a gather-scatter graph to handle the view operations. There is a simple slice test added, which works.